src/Adapter/Employee/ContextEmployeeProvider.php

```diff
@@ -35,14 +35,14 @@
 final class ContextEmployeeProvider implements ContextEmployeeProviderInterface
 {
     /**
-     * @var Employee
+     * @var Employee|null
      */
     private $contextEmployee;
 
     /**
-     * @param Employee $contextEmployee
+     * @param Employee|null $contextEmployee
      */
-    public function __construct(Employee $contextEmployee)
```

Comment:
how about ``Employee $contextEmployee = null```instead?

Reply:
In my case I'm collecting the hooks list so I have to call all the services and some of them will probably never be used from command line. In any case I get the error when this one is called.

If I remove this **null** usage here then there are only one possible solution right now I can think of:

In my console command I set fake employee `Context::getContext()->employee = new Employee(1)`.

Do you think its better?

Reply:
I think it's less worse.. sadly
We already have trouble create clean services, so if we start changing their classes because in some context we don't have the expected value it gonna get uglier and uglier You shouldn't change the class behaviour but rather its service definition

The risk here is that allowing a null employee for this CLI context could cause bugs in other context because a non expected null employee will have been allowed

It's not clean to init the context in your command but at least it limits the scope where you have to do dirty modifications.. I had the same issue in https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Command/GenerateMailTemplatesCommand.php because without the employee the hooks are not dispatched

Reply:
This is a bit related to https://github.com/PrestaShop/PrestaShop/issues/12918

Reply:
done according to @jolelievre  suggestion  :+1:

Reply:
> The risk here is that allowing a null employee for this CLI context could cause bugs in other context because a non expected null employee will have been allowed

@jolelievre I think that `new Employee(42)` is the same as `new Employee()`. If that's the case, I'd prefer not to use random constant `42` and avoid confusion.

There may also be employee with id 42, then the results may be totally unexpected and hard to debug imo.